### ISSUE TITLE
Duplicates

### ISSUE THREAD

**Issue by scottrobertson** [trakt/Plex-Trakt-Scrobbler#175]
Hey,

Getting duplicates in trakt all of a sudden. Running the latest (develop) version.

Any ideas?

Let me know if you need any logs etc

My Settings (if it helps):

![Settings](https://upload.scottrobertson.me/dakcDcW565vfP.png)

**Comment by fuzeman** [comment]
Duplicates can happen while the trakt.tv API is under heavy load, could you upload the plugin log file to confirm this though?

See https://github.com/fuzeman/Plex-Trakt-Scrobbler/wiki/Support#log-files for the location of the log files.

**Comment by scottrobertson** [comment]
I fixed this by using master, but since you merged develop, this has started happening again. Getting logs now

**Comment by scottrobertson** [comment]
https://gist.github.com/scottrobertson/27abaf20b32d9dac4db2

The pause at `00:01:32` was genuine.

**Comment by fuzeman** [comment]
Could you provide a log with all the "Logging" options set to "DEBUG", it's a bit hard to see why the duplication occurred from that log file..

**Comment by scottrobertson** [comment]
@fuzeman sorry about the late reply. This seems to have been fixed with the an update lately? No longer getting duplicates.

**Comment by scottrobertson** [comment]
Sorry, i thought this was sorted, but it's not. I will need to enable debug logging and try and get you a lot.

I turned scrolling off, and this still happened. This makes me think it happens during the sync processes.

**Comment by fuzeman** [comment]
I'm pretty sure this is caused by retrying requests on 5xx errors (when trakt.tv has actually processed the request, but still returned an error).

I'll disable request retrying in the next release *(will post a comment here when it's released)*.

**Comment by scottrobertson** [comment]
@fuzeman yeah that would make sense. Are they aware of this issue?

It may correlate with the fact that I am seeing some shows on my history page, but they are not marked as watched.

**Comment by fuzeman** [comment]
I've released **v0.9.1.22**: https://forums.plex.tv/index.php/topic/102818-rel-trakt/?p=884518, let me know if you find any more duplicates.

trakt have been making improvements to API stability over the last couple of months, this issue appears to just come back during peak times.

**Comment by cspot78** [comment]
had a quad update today, http://pastebin.com/nrssLQhv

**Comment by fuzeman** [comment]
This is the likely cause:
```
2015-03-22 12:54:28,230 (113bd9000) :  INFO (logging_handler:46) - [plugin.pts.scrobbler] [ST 98] First Time on the Front Line ('tvdb', '281406') stopped, watching status cancelled [u: 'cwarren78', c: 'Chucks-Mac-mini', s: 'TV Shows', a: '192.168.1.66']
2015-03-22 12:54:52,246 (1143df000) :  WARNING (logging_handler:46) - [requests.packages.urllib3.connectionpool] Retrying (Retry(total=2, connect=None, read=None, redirect=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='api-v2launch.trakt.tv', port=443): Read timed out. (read timeout=24)",)': /scrobble/stop
2015-03-22 12:54:52,247 (1143df000) :  INFO (logging_handler:46) - [requests.packages.urllib3.connectionpool] Starting new HTTPS connection (93): api-v2launch.trakt.tv
2015-03-22 12:55:16,376 (1143df000) :  WARNING (logging_handler:46) - [requests.packages.urllib3.connectionpool] Retrying (Retry(total=1, connect=None, read=None, redirect=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='api-v2launch.trakt.tv', port=443): Read timed out. (read timeout=24)",)': /scrobble/stop
2015-03-22 12:55:16,376 (1143df000) :  INFO (logging_handler:46) - [requests.packages.urllib3.connectionpool] Starting new HTTPS connection (94): api-v2launch.trakt.tv
2015-03-22 12:55:40,506 (1143df000) :  WARNING (logging_handler:46) - [requests.packages.urllib3.connectionpool] Retrying (Retry(total=0, connect=None, read=None, redirect=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='api-v2launch.trakt.tv', port=443): Read timed out. (read timeout=24)",)': /scrobble/stop
```

Should have another release today with more duplication fixes. *(watch this thread to see when it's released)*

--------------------------------

trakt has some performance improvements coming which should fix the root cause of these problems, this was posted yesterday:
> I just pushed some big optimization updates to staging.trakt.tv for the /sync/history and /sync/collection methods. It should be significantly faster and once deployed to production should have a big performance increase, and in turn help stabilize the servers.

**Comment by fuzeman** [comment]
I've released **v0.9.1.23**: https://forums.plex.tv/index.php/topic/102818-rel-trakt/?p=887021, fixes a couple more duplication bugs.

**Comment by fuzeman** [comment]
Haven't heard any report of duplicates (or seen any myself) after the latest release, closing this issue.
